src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseFVM.cpp

```diff
@@ -586,18 +586,26 @@ void CompositionalMultiphaseFVM::applySystemSolution( DofManager const & dofMana
 
   forMeshTargets( domain.getMeshBodies(), [&]( string const &,
                                                MeshLevel & mesh,
-                                               arrayView1d< string const > const & )
+                                               arrayView1d< string const > const & regionNames )
   {
-    std::map< string, string_array > fieldNames;
-    fieldNames["elems"].emplace_back( extrinsicMeshData::flow::deltaPressure::key() );
-    fieldNames["elems"].emplace_back( extrinsicMeshData::flow::deltaGlobalCompDensity::key() );
-
+    std::vector< SyncFieldsID > fieldsToBeSync;
     if( m_isThermal )
     {
-      fieldNames["elems"].emplace_back( extrinsicMeshData::flow::deltaTemperature::key() );
+      fieldsToBeSync.emplace_back( SyncFieldsID( FieldLocation::Elem,
+                                                 { extrinsicMeshData::flow::deltaPressure::key(),
+                                                   extrinsicMeshData::flow::deltaGlobalCompDensity::key(),
+                                                   extrinsicMeshData::flow::deltaTemperature::key() },
+                                                 regionNames ) );
+    }
+    else
+    {
+      fieldsToBeSync.emplace_back( SyncFieldsID( FieldLocation::Elem,
+                                                 { extrinsicMeshData::flow::deltaPressure::key(),
+                                                   extrinsicMeshData::flow::deltaGlobalCompDensity::key() },
+                                                 regionNames ) );
     }
```

Comment:
```suggestion
    SyncFieldsID tmp( FieldLocation::Elem, { extrinsicMeshData::flow::deltaPressure::key(), extrinsicMeshData::flow::deltaGlobalCompDensity::key() }, regionNames );
    if( m_isThermal )
    {
        tmp.fieldNames.push_back( extrinsicMeshData::flow::deltaTemperature::key() );
    }
    std::vector< SyncFieldsID > const fieldsToBeSync{ tmp };
```

Reply:
WE could also just emplace back a  new `SyncToBeField` entry and get rid of the temporary.